internal/handler/order_handler.go

```diff
@@ -0,0 +1,93 @@
+package handler
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/golangTroshin/gophermat/internal/middleware"
+	"github.com/golangTroshin/gophermat/internal/service"
+)
+
+type OrderHandler struct {
+	orderService service.OrderService
+}
+
+type OrderResponse struct {
+	Number     string  `json:"number"`
+	Status     string  `json:"status"`
+	Accrual    float64 `json:"accrual,omitempty"`
+	UploadedAt string  `json:"uploaded_at"`
+}
+
+func NewOrderHandler(orderService service.OrderService) *OrderHandler {
+	return &OrderHandler{orderService: orderService}
+}
+
+func (h *OrderHandler) UploadOrder(w http.ResponseWriter, r *http.Request) {
+	userID, ok := r.Context().Value(middleware.UserIDContextKey).(uint)
+	if !ok {
+		http.Error(w, "Invalid request format", http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil || len(body) == 0 {
+		http.Error(w, "Invalid request format", http.StatusBadRequest)
+		return
+	}
+	orderNumber := string(body)
+
+	if !h.orderService.ValidateOrderNumber(orderNumber) {
+		http.Error(w, "Invalid order number", http.StatusUnprocessableEntity)
+		return
+	}
+
+	err = h.orderService.CreateOrder(userID, orderNumber)
+	if err != nil {
+		if err == service.ErrOrderExistsSameUser {
+			w.WriteHeader(http.StatusOK)
+			return
+		} else if err == service.ErrOrderExistsDifferentUser {
+			http.Error(w, "Order already uploaded by another user", http.StatusConflict)
+			return
+		}
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
```

Comment:
Здесь можно использовать либо `errors.Is(...)` вместо `==`, если ты заинтересован в проверке конкретного типа ошибки, либо `errors.As(...)`  если тебе получить ошибку конкретного типа из цепочки ошибок.
```suggestion
		if errors.Is(err, service.ErrOrderExistsSameUser) {
			w.WriteHeader(http.StatusOK)
			return
		} else if errors.Is(err, service.ErrOrderExistsDifferentUser) {
			http.Error(w, "Order already uploaded by another user", http.StatusConflict)
			return
		}
		http.Error(w, "Internal server error", http.StatusInternalServerError)
```